dd-trace-core/src/main/java/datadog/trace/common/sampling/SpanSamplingRules.java

```diff
@@ -5,200 +5,230 @@
+  private final List<Rule> rules;
 
-  public static final class Rule {
-    public static Rule create(
-        String service, String name, Double sampleRate, Integer maxPerSecond) {
-      if (sampleRate == null) {
-        sampleRate = 1.0;
-      } else if (sampleRate < 0 || sampleRate > 1.0) {
-        logError(
-            service,
-            name,
-            Double.toString(sampleRate),
-            maxPerSecond == null ? null : Double.toString(maxPerSecond),
-            "sample_rate must be between 0.0 and 1.0");
-        return null;
-      }
-      if (maxPerSecond == null) {
-        maxPerSecond = Integer.MAX_VALUE;
-      } else if (maxPerSecond < 1) {
-        logError(
-            service,
-            name,
-            Double.toString(sampleRate),
-            Double.toString(maxPerSecond),
-            "max_per_second must be greater than zero");
-        return null;
-      }
-      return new Rule(service, name, sampleRate, maxPerSecond);
+  public SpanSamplingRules(List<Rule> rules) {
+    this.rules = rules;
+  }
+
+  public static SpanSamplingRules deserialize(String json) {
+    SpanSamplingRules result = EMPTY;
+    try {
+      result = filterOutNullRules(LIST_OF_RULES_ADAPTER.fromJson(json));
+    } catch (Throwable ex) {
+      log.error("Couldn't parse Span Sampling Rules from JSON: {}", json, ex);
     }
+    return result;
+  }
 
-    public static Rule create(String service, String name, String sampleRate, String maxPerSecond) {
-      double sampleRateParsed = 1.0;
-      if (sampleRate != null) {
-        try {
-          sampleRateParsed = Double.parseDouble(sampleRate);
-        } catch (NumberFormatException ex) {
-          logError(
-              service,
-              name,
-              sampleRate,
-              maxPerSecond,
-              "sample_rate must be a number between 0.0 and 1.0");
-          return null;
-        }
-      }
-      Integer maxPerSecondParsed = null;
-      if (maxPerSecond != null) {
-        try {
-          maxPerSecondParsed = Integer.parseInt(maxPerSecond);
-        } catch (NumberFormatException ex) {
-          logError(
-              service, name, sampleRate, maxPerSecond, "max_per_second must be greater than zero");
-          return null;
-        }
-      }
-      return create(service, name, sampleRateParsed, maxPerSecondParsed);
+  public static SpanSamplingRules deserializeFile(String jsonFile) {
+    SpanSamplingRules result = EMPTY;
+    try (JsonReader reader = JsonReader.of(Okio.buffer(Okio.source(new File(jsonFile))))) {
+      result = filterOutNullRules(LIST_OF_RULES_ADAPTER.fromJson(reader));
+    } catch (FileNotFoundException e) {
+      log.warn("Span sampling rules file {} doesn't exit", jsonFile);
+    } catch (IOException e) {
+      log.error("Couldn't read Span sampling rules file {}.", jsonFile, e);
+    } catch (Throwable ex) {
+      log.error("Couldn't parse Span Sampling Rules from JSON file {}.", jsonFile, ex);
```

Comment:
```suggestion
      log.error("Couldn't parse Span sampling rules from JSON file {}.", jsonFile, ex);
```
just making the case match the previous lines...